compiler/src/iree/compiler/Dialect/VM/Conversion/StandardToVM/ConvertStandardToVM.cpp

```diff
@@ -926,9 +926,11 @@ class FPToSIOpConversion : public OpConversionPattern<arith::FPToSIOp> {
       ConversionPatternRewriter &rewriter) const override {
     auto srcType = srcOp.getIn().getType();
     auto dstType = srcOp.getResult().getType();
+    auto resultType = getTypeConverter()->convertType(dstType);
     if (srcType.isF32()) {
-      if (dstType.isSignlessInteger(32) || dstType.isSignedInteger(32)) {
-        auto resultType = getTypeConverter()->convertType(dstType);
+      // This uses the resultType rather than dstType as any truncation
+      // required will be handled via interpretation by consumer.
+      if (resultType.isSignlessInteger(32) || resultType.isSignedInteger(32)) {
         rewriter.replaceOpWithNewOp<IREE::VM::CastF32SI32Op>(
             srcOp, resultType, adaptor.getOperands()[0]);
```

Comment:
since you're touching the code around this area, could you also update `adaptor.getOperands()[0]` to `adaptor.getIn()`?

Reply:
I was referring l.935, but it's still good that you fix the other part..